azure-toolkit-libs/azure-toolkit-appservice-lib/src/main/java/com/microsoft/azure/toolkits/appservice/model/Runtime.java

```diff
@@ -34,6 +37,91 @@
     public static final Runtime DOCKER = new Runtime(OperatingSystem.DOCKER, null, null);
 
+    public static Runtime getRuntime(OperatingSystem operatingSystem, WebContainer webContainer, JavaVersion javaVersion) {
+        return values().stream()
+                .filter(runtime -> runtime.operatingSystem == operatingSystem)
+                .filter(runtime -> runtime.webContainer == webContainer)
+                .filter(runtime -> runtime.javaVersion == javaVersion)
+                .findFirst().orElse(new Runtime(operatingSystem, webContainer, javaVersion));
+    }
+
+    public static Runtime getRuntimeFromWebApp(WebAppBase webAppBase) {
+        if (StringUtils.startsWithIgnoreCase(webAppBase.linuxFxVersion(), "docker")) {
+            return DOCKER;
+        }
+        return webAppBase.operatingSystem() == com.azure.resourcemanager.appservice.models.OperatingSystem.WINDOWS ?
+                getRuntimeFromWindowsWebApp(webAppBase) : getRuntimeFromLinuxWebApp(webAppBase);
+    }
+
+    public static Runtime getRuntimeFromLinuxWebApp(WebAppBase webAppBase) {
+        if (StringUtils.isEmpty(webAppBase.linuxFxVersion())) {
+            return null;
+        }
+        final String linuxFxVersion = webAppBase.linuxFxVersion().replace("|", " ");
+        return getRuntimeFromLinuxFxVersion(linuxFxVersion);
+    }
+
+    public static Runtime getRuntimeFromLinuxFxVersion(String linuxFxVersion) {
+        final JavaVersion javaVersion = StringUtils.containsIgnoreCase(linuxFxVersion, "java11") ? JavaVersion.JAVA_11 : JavaVersion.JAVA_8;
+        final WebContainer webContainer = WebContainer.values().stream()
+                .filter(container -> StringUtils.containsIgnoreCase(linuxFxVersion, container.getValue()))
+                .findFirst().orElse(null);
+        return getRuntime(OperatingSystem.LINUX, webContainer, javaVersion);
+    }
+
+    public static Runtime getRuntimeFromWindowsWebApp(WebAppBase webAppBase) {
+        if (webAppBase.javaVersion() == null || StringUtils.isAnyEmpty(webAppBase.javaContainer(), webAppBase.javaContainerVersion())) {
+            return null;
+        }
+        final JavaVersion javaVersion = JavaVersion.values().stream()
+                .filter(version -> StringUtils.equals(webAppBase.javaVersion().toString(), version.getValue()))
+                .findFirst().orElse(null);
+        final String javaContainer = String.join(webAppBase.javaContainer(), " ", webAppBase.javaContainerVersion());
+        final WebContainer webContainer = StringUtils.equalsIgnoreCase(webAppBase.javaContainer(), "java") ? WebContainer.JAVA_SE :
+                WebContainer.values().stream()
+                        .filter(container -> StringUtils.equals(javaContainer, container.getValue()))
+                        .findFirst().orElse(null);
+        return getRuntime(OperatingSystem.WINDOWS, webContainer, javaVersion);
+    }
+
+    public RuntimeStack convertToServiceRuntimeStackModel() {
```

Comment:
what is the difference between `RuntimeStack` and `Runtime`, are both of them necessary?

Reply:
This is the utils convert library model to sdk model, will move to manager utils later